### ISSUE TITLE
Setting window position before creating it

### ISSUE THREAD

**Issue by NoorWachid** [glfw/glfw#1747]
In the example video below I tried to centerize window positions. Because in GLFW moving is done after creation the window looks jumping across default position given by window manager [I tried mutter, openbox, xfwm4]. On the other-hand in SDL2 determines the window position beforehand. 
https://streamable.com/y9jntf

**Comment by elmindreda** [comment]
Yup, sorry, that's an oversight in the GLFW 3 API. You can work around it with a bit of extra code.

```c
glfwWindowHint(GLFW_VISIBLE, false);
window = glfwCreateWindow(...);
glfwSetWindowPos(window, x, y);
glfwShowWindow(window);
```
